Ports/gcc/patches/gcc-support-mold-linker.patch

```diff
@@ -0,0 +1,129 @@
+From 37f6a0c24bb07efb29a2222d984e4d793c165aef Mon Sep 17 00:00:00 2001
+From: Martin Liska <mliska@suse.cz>
+Date: Tue, 21 Dec 2021 17:43:55 +0100
+Subject: [PATCH] Support ld.mold linker.
```

Comment:
This one is a bit obvious, but it might be time we add a `ReadMe.md` for Toolchain patches.

Reply:
I promise I'm going to clean up the toolchain patches very soon. I'd rather wait for that, as the existing patches are *hard to describe*. The best description I can come up with "A random mismash of stuff".

Reply:
Sure, just trying to gain concensus on future direction. 😁